tests/net/socket/tls_configurations/README.rst

```diff
@@ -0,0 +1,65 @@
+.. zephyr:code-sample:: socket-tls-configurations
+   :name: TLS 1.2 and 1.3 configurations with sockets
+   :relevant-api: bsd_sockets tls_credentials
+
+   Showcase how to up a socket-tls connection with different TLS configurations
+   and test it against an OpenSSL server.
+
+Overview
+********
+
+This sample showcases various minimal Mbed TLS configurations to properly set up
```

Comment:
This is not a sample anymore. Plus, is this documentation accessible anywhere? It could even be removed completely.

Reply:
Back to when this really was a sample it was (I checked), but I did not re-checked since I moved it to the `tests`. Once I can have the HTML build fine again, I will check again.
I think it makes sense to have a small README just to explain what this test is about. Do you think it's worth it? Or is it clear from other comments in the test?

Reply:
I don't think it's really worth it. The code speaks for itself (especially the `testcase.yaml` file), plus I don't think we have documentation for tests?